### DISCUSSION TITLE
SX1276/RFM95 FSK Beacon Mode.

### DISCUSSION THREAD

**Discussion by abhijit-031991** [jgromes/RadioLib#517] [Q&A]
Hello, 

I have a project that needs to send out an FSK signal (2 bytes) at preset intervals. As I can see in the datasheet, there is a use case for 'beacon mode' using the in-built timers and the top sequencer. As far as I can tell, the following things need to be done.

1. Set _beaconOn_ bit in the _RegPacketConfig2_ register.
2. Set the relevant sequencer bits in the _RegSeqConfig1_, _RegTimerResol_, _RegTimer1Coef_ and _RegTimer2Coef_

I see in Module.h that there are several SPI methods. However, I'm not sure how to use them.
I would like to experiment with setting these registers. 
While I know this can be achieved by MCU control. The power specs indicate this to be a far more power-efficient
[DS_SX1276-7-8-9_W_APP_V7.pdf](https://github.com/jgromes/RadioLib/files/8646997/DS_SX1276-7-8-9_W_APP_V7.pdf)
 method.

I am rather new at coding and any help would be greatly appreciated!

**Comment by jgromes** [comment]
This feature is not implemented in RadioLib directly, however, you can use the low-level SPI interface to program the SPI registers of the SX127x device. You will have the enable the macro `RADIOLIB_LOW_LEVEL` in BuildOpt.h. Once you do that, you will be able to use the SPI get/set etc. methods like so:

```c++
SX1278 radio = new Module(10, 2, 9, 3);

(...)

// set register RADIOLIB_SX127X_REG_PACKET_CONFIG_2 to 0x12
radio._mod->SPIsetRegValue(RADIOLIB_SX127X_REG_PACKET_CONFIG_2, 0x12);
```

I can't really comment on the power efficiency, as I haven't used the beacon mode in SX127x sequencer.

**Reply by abhijit-031991** [reply]
Hi!

Thank you for your super quick reply!! I'm sure this should be super helpful!
